### PR TITLE
GCS: Fix compiler warnings

### PR DESCRIPTION
**Pull request by peabody124**
Suppress compile warnings on GCS and enable -Werror for linux

### CODE REVIEW

**Comment by peabody124** [comment]
See https://github.com/d-ronin/dRonin/pull/59/commits and https://github.com/d-ronin/dRonin/pull/85/commits

**Comment by kubark42** [comment]
LGTM, except I'm not sure we should have the default build be with error on. I like Jenkins catching that, but it's a bit of a PITA during early development.

File: ground/gcs/src/libs/libcrashreporter-qt/breakpad/client/linux/handler/exception_handler.cc [hunk 1]
```diff
@@ -385,6 +385,8 @@ int ExceptionHandler::ThreadEntry(void *arg) {
 // This function runs in a compromised context: see the top of the file.
 // Runs on the crashing thread.
 bool ExceptionHandler::HandleSignal(int sig, siginfo_t* info, void* uc) {
+  (void)sig;
```
Comment (kubark42) [inline ground/gcs/src/libs/libcrashreporter-qt/breakpad/client/linux/handler/exception_handler.cc]: indentation

**Comment by peabody124** [comment]
@kubark42 we could have it only apply in the release configuration. I think that is what jenkins builds anyway.

**Comment by kubark42** [comment]
@peabody124, yeah, I think that'd be great. I'll do that in a subsequent PR, it only affects Linux anyway.

**Pull request merged**